src/core/core.controller.js

```diff
@@ -1018,23 +1018,24 @@ class Chart {
 	/**
 	 * Handle an event
 	 * @param {IEvent} e the event to handle
+	 * @param {boolean} [replay] - true if the event was replayed by `update`
 	 * @return {boolean} true if the chart needs to re-render
 	 * @private
 	 */
-	_handleEvent(e) {
+	_handleEvent(e, replay) {
 		const me = this;
-		const options = me.options || {};
+		const options = me.options;
 		const hoverOptions = options.hover;
+		// If the event is replayed from `update`, we should evaluate with the final positions.
```

Comment:
Can you explain more why the position from `update` might not be the final position?

Reply:
Element's properties are its current state ("view" in 2.9) and the animations contain the target ("model" in 2.9). So when animations are enabled, after update and before first animation frame, elements x/y are exactly the same than before update. Only the animations were updated. The Animator then updates the elements in each frame.